### PR TITLE
WIP: Rudimentary text + URL + title searching.

### PR DESCRIPTION
**Pull request by rnewman**

### CODE REVIEW

**Comment by rnewman** [comment]
@bgrins, @ncalexan: save some page content from the menu and try this out?

**Comment by rnewman** [comment]
It's expected not to pass lint yet, because we do dangerous things.

File: app/services/storage.js [hunk 1]
```diff
@@ -373,6 +374,77 @@ export class ProfileStorage {
     return out;
   }
 
+  /**
+   * Search for places that match title, URL, or saved content
+   * since the specified timestamp.
+   *
+   * Snippets in the returned results are guaranteed to be HTML-escaped.
```
Comment (ncalexan) [inline app/services/storage.js]: What does this mean?  That they can be interpolated into HTML as values?  Would other consumers find this irritating?

File: app/shared/types.js [hunk 2]
```diff
@@ -23,3 +23,11 @@ export type ReadabilityResult = {
   length: number,
   excerpt: string,
 };
+
+export type AwesomebarMatch = {
```
Comment (ncalexan) [inline app/shared/types.js]: I feel this pattern is not helping.  Maybe we can just use ES6 class definitions for this?

File: app/ui/browser/views/navbar/location.jsx [hunk 3]
```diff
@@ -220,17 +227,41 @@ export class Location extends Component {
...
-        }}
-        style={this.state.focusedResultIndex === i ? { background: 'red' } : null}>
-        {completion}</div>
-      ));
+    const renderRow = (completion, i) => {
+      // We get safe, decorated (<b>foo</b>) HTML from the database.
+      const snippet = completion.snippet ? (
+        <div
+          className={LOCATION_BAR_RESULTS_SNIPPET_STYLE}
+          dangerouslySetInnerHTML={{          // eslint-disable-line react/no-danger
```
Comment (ncalexan) [inline app/ui/browser/views/navbar/location.jsx]: How can this rule not pun on "middle name"?

**Comment by ncalexan** [comment]
This is looking pretty great!

File: app/services/storage.js (hunk 1 above)
Comment (rnewman) [inline app/services/storage.js]: Perhaps another way of phrasing this is that the results are valid HTML fragments.

File: app/ui/browser/views/navbar/location.jsx (hunk 3 above)
Comment (rnewman) [inline app/ui/browser/views/navbar/location.jsx]: I'm waiting for a trolling rule whose name is more than 100 characters long.

File: app/services/storage.js (hunk 1 above)
Comment (rnewman) [inline app/services/storage.js]: And yes, the idea is that this API is generating HTML like it's 2007. The alternative is that we return some super awesome result format like:

```
[{t: "… some text containing "},
 {m: "a match"},
 {t: " then some more text."}]
```

File: app/shared/types.js (hunk 2 above)
Comment (rnewman) [inline app/shared/types.js]: I assume we'll be ditching this if we get rid of Flow…

**Pull request closed without merging**